src/main/java/emu/grasscutter/game/quest/content/ContentClearGroupMonster.java

```diff
@@ -3,14 +3,17 @@
 import emu.grasscutter.data.excels.QuestData;
 import emu.grasscutter.game.quest.GameQuest;
 import emu.grasscutter.game.quest.QuestValueContent;
-import emu.grasscutter.game.quest.enums.QuestContent;
 import lombok.val;
 
-@QuestValueContent(QuestContent.QUEST_CONTENT_CLEAR_GROUP_MONSTER)
+import static emu.grasscutter.game.quest.enums.QuestContent.QUEST_CONTENT_CLEAR_GROUP_MONSTER;
+
+@QuestValueContent(QUEST_CONTENT_CLEAR_GROUP_MONSTER)
 public class ContentClearGroupMonster extends BaseContent {
+
     @Override
     public boolean execute(GameQuest quest, QuestData.QuestContentCondition condition, String paramStr, int... params) {
         val groupId = condition.getParam()[0];
-        return quest.getOwner().getScene().getScriptManager().hasClearedGroupMonsters(groupId);
+
+        return params[0] == groupId && quest.getOwner().getScene().getScriptManager().isClearedGroupMonsters(groupId);
```

Comment:
Probably better to check only the fixed data, for everything except for lua notify, im actually on the way to remove the usage of the event params, and only compare to the actual data.
Makes it easier to check if something is already completed, when the quest starts.

Reply:
Not really sure, but you can change it on merge